### DISCUSSION TITLE
Discussion: Fluent v2 NavigationView Top Selection Indicator too Small

### DISCUSSION THREAD

**Discussion by robloo** [microsoft/microsoft-ui-xaml#7427] [General]
# Discussion: Fluent v2 Header Underlines too Small

Personally, for top header mode, I think the NavigationView selection indicator is a poor design choice. It makes sense for vertically stacked items but not for horizontal tab-like interfaces.

 1) Can this be changed back to what it was in Fluent v1? As your design team intentionally made this change I somewhat doubt it will change back no matter how bad it looks.
 2) Is there a quick way developers can switch this back without re-templating? I didn't see any lightweight styling resources for it.

Fluent v2: (poor selection indicator design)

![image](https://user-images.githubusercontent.com/17993847/145731887-51336940-061a-4d13-86f2-054883c05c7d.png)

Fluent v1: (better selection indicator design)

![image](https://user-images.githubusercontent.com/17993847/145731917-342791ed-c378-4469-970c-62c547937308.png)

**Comment by fitojb** [comment]
I like the smaller pill, but the selection visual indication could benefit from a sligtly darker background (or clearer, in the dark theme), just like vertical lists do.

**Comment by Panda-Sharp** [comment]
We're in the same situation, we would really love to make it bigger with no need to re-template it, a lightweight option would be great!

**Comment by StephenLPeters** [comment]
This is by the new design, but exposing it as a light weight resource is a good idea. @karenbtlai Do you know where the change would be?

**Comment by marcelwgn** [comment]
@StephenLPeters Is this ready for community work then, i.e. should a new lightweight styling resource should be introduced for this?

**Comment by harvinders** [comment]
@StephenLPeters A light weight resource would be really good. Microsoft may feel the default is solving the needs, but our customers are reporting difficulty figuring out which tab is selected easily. They really had to look for it, rather than just taking a cursory glance.

**Comment by karkarl** [comment]
I can take a look. Let me also bring this issue up with design.

**Comment by robloo** [comment]
@karenbtlai Here is the place where this is set. It has been re-templated for now in the app to solve the issue. A fix in WinUI likely needs two new resources for Width and HorizontalAlignment.

https://github.com/microsoft/microsoft-ui-xaml/blob/01eea310f89b72109a36de6537e3357fb2421556/dev/NavigationView/NavigationView_rs1_themeresources.xaml#L1209

Design fix looks like:

```xaml
<Grid
    x:Name="SelectionIndicatorGrid"
    Margin="16,0,16,4"
    VerticalAlignment="Bottom"
    HorizontalAlignment="Stretch">
    <Rectangle
        x:Name="SelectionIndicator"
        Height="3"
        MinWidth="16"
        HorizontalAlignment="Stretch"
        Fill="{ThemeResource NavigationViewSelectionIndicatorForeground}"
        Opacity="0"
        RadiusX="2"
        RadiusY="2" />
</Grid>
```
